### PR TITLE
feat: Write data

### PR DESCRIPTION
**Pull request by stanleyjones**
## Description

The UI work and beginnings of API work for writing arbitrary data to the KVStore.

## Related Issue

Resolve liftedinit/roadmap#16

## Testing / Breaking Changes

Testing manually for now. No breaking changes.

## Screenshots (if applicable)

**NO BUTTON FOR ANONYMOUS**
Screenshot 2022-12-15 at 1 26 46 PM

**BUTTON**
Screenshot 2022-12-15 at 1 26 55 PM

**MODAL FORM**
Screenshot 2022-12-15 at 1 27 00 PM

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
     "@emotion/react": "^11.10.4",
     "@emotion/styled": "^11.10.4",
     "@fontsource/rubik": "^4.5.11",
-    "@liftedinit/many-js": "github:liftedinit/many-js",
+    "@liftedinit/many-js": "github:stanleyjones/many-js#servers-and-services",
```
Comment (stanleyjones) [inline package.json:10]: Need the branch from https://github.com/liftedinit/many-js/pull/85.

File: package.json [hunk 2]
```diff
@@ -67,5 +67,5 @@
       "last 1 safari version"
     ]
   },
-  "proxy": "http://localhost:8000"
+  "proxy": "http://localhost:8010"
```
Comment (stanleyjones) [inline package.json:70]: The KVStore runs locally at this port.

File: src/features/services/components/data-settings.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+export function DataSettings() {
+  const account = useAccountsStore((s) => s.byId.get(s.activeId));
+  const { isOpen, onOpen, onClose } = useDisclosure();
+  return (
+    <>
+      <Box p={6} bg="white" mt={9} boxShadow="xl">
+        <Center>
+          <pre>PLACEHOLDER FOR liftedinit/roadmap#17</pre>
+        </Center>
```
Comment (stanleyjones) [inline src/features/services/components/data-settings.tsx:14]: @mjsorribas Your table can go here. I'm trying to preemptively avoid merge conflicts by leaving space.

File: src/features/services/queries.ts [hunk 4]
```diff
@@ -89,3 +90,22 @@ export function useCreateToken() {
       }),
   });
 }
+
+export function usePutValue() {
+  // eslint-disable-next-line
+  const [_, network] = useNetworkContext();
+  const queryClient = useQueryClient();
+  const { url, id } = network!;
+  const kv = new KeyValueService(url, id);
```
Comment (stanleyjones) [inline src/features/services/queries.ts:99]: Using the new service. But of course it's not quite working yet.

File: src/features/services/queries.ts [hunk 5]
```diff
@@ -89,3 +90,22 @@ export function useCreateToken() {
...
+
+  return useMutation<unknown, Error, PutValueInputs>({
+    mutationFn: async (inputs: PutValueInputs) => {
+      await kv.put(inputs);
+    },
+    onSuccess: () =>
+      // @TODO: Invalidate the query from liftedinit/roadmap#17
+      queryClient.invalidateQueries({
+        queryKey: ["", network?.url],
```
Comment (stanleyjones) [inline src/features/services/queries.ts:108]: When we save, we want to invalidate the query from the "read data" story so it's automatically fetched again.

**Comment by stanleyjones** [comment]
This "should" work but it's currently crashing my local KVStore back-end. Going to investigate.

File: package.json (hunk 2 above)
Comment (mjsorribas) [reply package.json:70]: All the stuff  runs over 8010? or we need to switch depends the service? Is not better used something like this:
```
"proxy": {
  "/api": {
    "target": "http://localhost:8000/"
    },
  "/kvstore": {
    "target": "http://localhost:8010/"
  }
}
```

**Comment by stanleyjones** [comment]
Working now. I'll get this into review shortly.

File: package.json (hunk 2 above)
Comment (stanleyjones) [reply package.json]: Unfortunately, you can just add additional endpoints into the `package.json` but I added the `http-proxy-middleware` package so we can customize proxies by endpoint.

File: package.json [hunk 6]
```diff
@@ -66,6 +67,5 @@
       "last 1 firefox version",
       "last 1 safari version"
     ]
-  },
-  "proxy": "http://localhost:8000"
```
Comment (stanleyjones) [inline package.json:70]: Removed from `package.json` but see `setupProxy.js`.

File: src/setupProxy.js [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+    createProxyMiddleware({
+      target: "http://localhost:8000",
+      changeOrigin: true,
+    })
+  );
+  app.use(
+    "/kvapi",
+    createProxyMiddleware({
+      target: "http://localhost:8010",
+      changeOrigin: true,
```
Comment (stanleyjones) [inline src/setupProxy.js:15]: NOTE: This approach will require setting up a new "network" in the dropdown with the URL of `/kvapi` before it works.

**Review by mjsorribas — APPROVED** [review]

**Pull request merged**